s3fs/tests/test_s3fs.py

```diff
@@ -93,6 +93,11 @@ def test_client_kwargs():
     assert s3.s3.meta.endpoint_url.startswith('http://foo')
 
 
+def test_config_kwargs():
+    s3 = S3FileSystem(config_kwargs={'signature_version': 's3v4'})
+    assert s3.connect(refresh=True).meta.config.signature_version == 's3v4'
```

Comment:
Aren't they the same? `self.s3 = self.connect()` in `S3FileSystem.__init__` -- I only call the function explicitly to pass `refresh=True`, without which it uses a cache from a previous `S3FileSystem` instance returns the wrong `signature_version`.